base/dcp_client.go

```diff
@@ -84,7 +84,7 @@ func NewDCPClient(ID string, callback sgbucket.FeedEventCallbackFunc, options DC
 		terminator:       make(chan bool),
 		doneChannel:      make(chan error, 1),
 		failOnRollback:   options.FailOnRollback,
-		checkpointPrefix: DCPCheckpointPrefixWithGroupID(groupID),
+		checkpointPrefix: DCPCheckpointPrefixWithGroupID(options.MetadataConfig.GroupID),
```

Comment:
I don't think GroupID makes (more) sense as a property of MetadataConfig, if it's only used for one type of metadata config (DCPMetadataDB).  I agree it would be nice to remove groupID as a standalone parameter on NewDCPClient.  I'd suggest one of the following:
- move groupID to DCPClientOptions
- remove groupID and add checkpointPrefix to DCPClientOptions, and rely on the caller to set when applicable

Reply:
Moved this up a level.